### ISSUE TITLE
Electron 11 stable-release for darwin x64 cannot run with rosetta2 on Apple Silicon

### ISSUE THREAD

**Issue by Pharlon** [electron/electron#26530]
### Preflight Checklist

* [ ☑️] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [ ☑️] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [ ☑️] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

We still need to work with Electron build for arch x86_64 under rosetta2, due to  pretty complex c++ addon which cannot easily rebuild for arm64. 
Every thing works fine until our mac  upgrade to macOS 11.0.1, the app is not supported any more

* **Electron Version:**
  * 11.0.0 stable
* **Operating System:**
  * Apple Silicon MacOS 11.0.1
* **Last Known Working Electron version:**
  * MacOS 11.0 beta

### Expected Behavior
works with rosetta2

### Actual Behavior
electron.app is not supported

### To Reproduce
run 'electron-v11.0.0-darwin-x64' directly on apple silicon with latest big sur

### Screenshots

![image](https://user-images.githubusercontent.com/19788989/99372664-76cd4b00-28fb-11eb-8880-9ca0e0c95931.png)

![image](https://user-images.githubusercontent.com/19788989/99372655-70d76a00-28fb-11eb-8e4f-e576ac81be8a.png)

**Comment by ckerr** [comment]
CC @MarshallOfSound @jkleinsc

**Comment by k4113h** [comment]
I'm having exactly the same issue: my app worked under Rosetta2 when still running Big Sur beta, but after upgrading to the official Big Sur release, the app doesn't start anymore and has the "no sign" in the icon. Command line:

```
% open MyApp.app 
The application cannot be opened for an unexpected reason, error=Error Domain=NSOSStatusErrorDomain Code=-10661 "(null)" UserInfo={_LSLine=3665, _LSFunction=_LSOpenStuffCallLocal}
```

**Comment by MarshallOfSound** [comment]
I can repro this, I'm assuming though that this is specific to the DTK and will not occur on M1 machines.  I'm just checking up on this though, will see what I find.

**Comment by MarshallOfSound** [comment]
Confirmed that this will work just fine on the M1 machines.  If you want to workaround this on your DTK ensure you launch the app from terminal directly. E.g.

```
./MyApp.app/Contents/MacOS/MyApp
```

**Comment by k4113h** [comment]
Thanks @MarshallOfSound! I can confirm that the workaround works in DTK.

**Comment by k4113h** [comment]
I updated to 11.0.3 and now the app works again normally. This was included in 11.0.2: https://github.com/electron/electron/pull/26570, might be related.
